core/runtime/common/trie_storage_provider_impl.cpp

```diff
@@ -174,7 +197,7 @@ namespace kagome::runtime {
   }
 
   outcome::result<void> TrieStorageProviderImpl::rollbackTransaction() {
-    if (transaction_stack_.empty()) {
+    if (transaction_stack_.size() <= 1) {
```

Comment:
Does the one also counts? 🤔

Reply:
There must always be `overlay`, at least 1 layer, because we should not mutate `backend` for `clear_prefix`.